CODE_OF_CONDUCT.md

```diff
@@ -0,0 +1,149 @@
+* Giving and gracefully accepting constructive feedback
+* Accepting responsibility and apologizing to those affected by our mistakes,
+  and learning from the experience
+* Focusing on what is best not just for us as individuals, but for the
+  overall community
+
+Examples of unacceptable behavior include:
+
+* The use of sexualized language or imagery, and sexual attention or
+  advances of any kind
+* Trolling, insulting or derogatory comments, and personal or political attacks
+* Public or private harassment
+* Publishing others' private information, such as a physical or email
+  address, without their explicit permission
+* Other conduct which could reasonably be considered inappropriate in a
+  professional setting
+
+## Enforcement Responsibilities
+
+Moderators are responsible for clarifying and enforcing our standards of
+acceptable behavior and will take appropriate and fair corrective action in
+response to any behavior that they deem inappropriate, threatening, offensive,
+or harmful.
+
+Moderators have the right and responsibility to remove, edit, or reject
+comments, commits, code, wiki edits, issues, and other contributions that are
+not aligned to this Code of Conduct, and will communicate reasons for moderation
+decisions when appropriate.
+
+Moderators are held to a higher standard than other community members. If a
+moderator creates an inappropriate situation, they should expect less leeway
+than others.
+
+## Scope
+
+This Code of Conduct applies within all community spaces, and also applies when
+an individual is officially representing the community in public spaces.
+Examples of representing our community include official IRC channels (#testing
+on irc.w3.org); GitHub repositories under the web-platform-tests organization;
+and the public-test-infra@w3.org mailing list.
+
+There may arise situations where both the WPT code of conduct and that of
+another organization (such as the WHATWG or W3C) may apply.
+For example, a WPT-focused meeting at
+[TPAC](https://www.w3.org/2002/09/TPOverview.html) would involve both the WPT
+code of conduct and the [W3C code of ethics and professional
+conduct](https://www.w3.org/Consortium/cepc/).
+In such situations we operate under all code of conducts involved.
+If you are placed in a situation where you feel this is inappropriate (e.g. if
+you believe the code of conducts involved contradict one another), please
+contact a [moderator](#moderators).
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported to the [moderators](#moderators).
+All complaints will be reviewed and investigated promptly and fairly.
+
+All moderators are obligated to respect the privacy and security of the
+reporter of any incident.
+
+Moderators will recuse themselves if they are directly involved in a report of
+a code of conduct violation.
+
+## Enforcement Guidelines
+
+Moderators will follow these Community Impact Guidelines in determining
```

Comment:
I think this being a set of options that the moderators may pick from according to the requirements of the system is an improvement over the strict consequence ladder.